File: backend_samples/backend/dotnet/Storage/AppSession.cs
namespace FaceLivenessAttestationBackendSample.Storage;

/// <summary>
/// App-owned session data (kept separate from the attestation library's session
/// record): the Face resource + key used to poll the liveness result. Never sent
/// to the browser.
/// </summary>
public sealed class AppSession
{
    public required string Resource { get; init; }
    public required string ApiKey { get; init; }
    /// <summary>"detectLiveness" or "detectLivenessWithVerify".</summary>
    public required string Action { get; init; }
}

/// <summary>Store for the app-owned <see cref="AppSession"/>, keyed by session id.</summary>
public interface IAppSessionStore
{
    Task<bool> SaveAsync(string sid, AppSession session);
    Task<AppSession?> GetAsync(string sid);
}

File: backend_samples/backend/java/src/main/java/com/azure/ai/vision/face/sample/store/AppSession.java
package com.azure.ai.vision.face.sample.store;

/**
 * App-owned per-session context, kept SEPARATE from the library's session
 * record: the Face credentials + action the sample uses to poll the liveness
 * result. Never sent to the browser.
 */
public record AppSession(String resource, String apiKey, String action) {
}
